actions.js

```diff
@@ -150,22 +236,20 @@ class instance extends instance_skel {
 					default: 'Hello World',
 				},
 			],
+			callback: function (action) {
+				self.doAction(action)
+			},
 		}
-		this.setActions(actions)
-	}
 
-	/**
-	 * Executes the provided action.
-	 *
-	 * @param {Object} action - the action to be executed
-	 * @access public
-	 * @since 1.0.0
-	 */
-	async action(action) {
-		const opt = action.options
+		self.setActionDefinitions(actions);
+	},
+
+	// Select action and make API call
+	doAction: function (act) {
```

Comment:
A suggestion (feel free to ignore or do at a later date):

I would recommend reworking this so that instead of this function taking in `act` it has `method` and `cmd` as parameters.
This will make it a lot easier to follow what each action actually does, and keeps the code a bit simpler.